mlx/backend/cuda/copy/copy_contiguous.cu

```diff
@@ -37,16 +71,23 @@ void copy_contiguous(
     int64_t out_offset) {
   dispatch_all_types(in.dtype(), [&](auto in_type_tag) {
     dispatch_all_types(out.dtype(), [&](auto out_type_tag) {
-      dispatch_bool(out.data_size() > UINT32_MAX, [&](auto large) {
+      dispatch_bool(out.data_size() > INT32_MAX, [&](auto large) {
```

Comment:
I think there is a bug here when using `int` that we should address before it gets to messy. In `get_2d_grid_dims_common` we use `UINT32_MAX` as the largest possible size for a grid element (see [here](https://github.com/ml-explore/mlx/blob/main/mlx/backend/common/utils.cpp#L151-L153)). So I think you could end up with a case where you overflow `int32`.

Reply:
In the Metal back-end we always use `UINT32_MAX` for the condition for contiguous elementwise ops and `uint` for the index type.. so we don't have this issue. But if we are going to use `int` for the index type then we need a version of that function which let's you specify the maximum allowed value.

Alternatively we could use `uint` for the index type with CUDA and rework the contiguous kernels so they don't rely on signed numbers.

Reply:
For example you could change which would also be a bit simpler and probably just as fast and allow for unsigned indexing:


```c++
  IdxT remaining = size - index * N_READS;
  if (remaining <= 0) {
    return;
  }

  if (remaining < N_READS) {
    for (int i = 0; i < remaining; ++i) {
      IdxT offset = index * N_READS + i;
      out[offset] = Op{}(a[offset], b[offset], c[offset]);
    }
  } else {
```

To 

```c++
  if (size >= (index + 1) * N_READS) {
    for (int i = index * N_READS; i < size; ++i) {
      out[i] = Op{}(a[i], b[i], c[i]);
    }
  } else {
```